ensime_shared/ensime.py

```diff
@@ -584,9 +584,13 @@ def handle_completion_info_list(self, call_id, payload):
         """Handler for a completion response."""
         completions = payload["completions"]
         self.log("handle_completion_info_list: in")
-        self.suggestions = [completion_to_suggest(c) for c in completions]
+        self.suggestions = [completion_to_suggest(c, self.format_type_params()) for c in completions]
         self.log("handle_completion_info_list: {}".format(self.suggestions))
 
+    def format_type_params(self):
+        """NOP as it's not available for Server 1.0"""
+        return lambda _: ""
+
```

Comment:
I was in doubt about implenting it as an abstract method. So you prefer to have a common class between versions and implement concrete classes for both versions, instead of make v2 override v1 behavior. Is that correct? It seems clearer to me.